lib/oli_web/controllers/api/automation_setup_controller.ex

```diff
@@ -0,0 +1,332 @@
+          - Archive is ingested and a project is created for that author.
+        If createSection is set:
+          - The project is published (if projectArchive was set)
+          - An educator account is created
+          - The educator account is linked to the author (if projectArchive was set)
+          - A section is created for that educator
+            - With the project created set (if projectArchive was set)
+        If createLearner is set:
+          - A learner account is created
+          - It's enrolled in the section (if it was created)
+      """,
+      type: :object,
+      properties: %{
+        project_archive: %Schema{
+          type: :file,
+          description: "Zip file containing a project export"
+        },
+        create_author: %Schema{
+          type: :boolean,
+          description: "Should an author account be created?"
+        },
+        create_section: %Schema{
+          type: :boolean,
+          description: "Should a section be set up that teaches this project?"
+        },
+        create_learner: %Schema{
+          type: :boolean,
+          description: "Should a learner be created?"
+        },
+        create_educator: %Schema{
+          type: :boolean,
+          description: "Should an educator account be created? Required if createSection is true"
+        }
+      },
+      required: []
+    })
+  end
+
+  @doc parameters: [],
+       security: [%{"bearer-authorization" => []}],
+       request_body:
+         {"Setup Request", "multipart/form-data",
+          OliWeb.Api.AutomationSetupController.AutomationSetupBody, required: true},
+       responses: %{
+         200 =>
+           {"Setup Response", "application/json",
+            OliWeb.Api.AutomationSetupController.AutomationSetupResponse}
+       }
+  def setup(conn, %{
+        "create_author" => create_author,
+        "create_learner" => create_learner,
+        "create_section" => create_section,
+        "create_educator" => create_educator,
+        "project_archive" => project_archive
+      }) do
+    if is_valid_api_key?(conn, &Oli.Interop.validate_for_automation_setup/1) do
+      case setup_data(
+             project_archive,
+             create_learner,
+             create_educator,
+             create_author,
+             create_section
+           ) do
+        {:error, reason} ->
+          error(conn, 400, reason)
+
+        {:ok, author, author_password, educator, educator_password, learner, learner_password,
+         project, section} ->
+          json(
+            conn,
+            %{
+              author: format_user(author, author_password),
+              educator: format_user(educator, educator_password),
+              learner: format_user(learner, learner_password),
+              project: format_project(project),
+              section: format_section(section),
+              success: true
+            }
+          )
+      end
+    else
+      error(conn, 403, "invalid api key")
```

Comment:
Some of our other API calls send a 400 response here, 403 seemed more semantically correct for an invalid api key.